src/sql/azdata.proposed.d.ts

```diff
@@ -1801,5 +1821,73 @@ declare module 'azdata' {
 
 		export interface Dialog extends LoadingComponentBase {
 		}
+
+		/**
+		 * Opens the error dialog with customization options provided.
+		 * @param options Dialog options to customize error dialog.
+		 * @returns Id of action button clicked by user, e.g. ok, cancel
+		 */
+		export function openCustomErrorDialog(options: IErrorDialogOptions): Thenable<string | undefined>;
+
+		/**
+		 * Provides dialog options to customize modal dialog content and layout
+		 */
+		export interface IErrorDialogOptions {
+			/**
+			 * Severity Level to identify icon of modal dialog.
+			 */
+			severity: MessageLevel;
+			/**
+			 * Title of modal dialog header.
+			 */
+			headerTitle: string;
+			/**
+			 * Message text to show on dialog.
+			 */
+			message: string;
+			/**
+			 * (Optional) Detailed message, e.g stack trace of error.
+			 */
+			messageDetails?: string;
+			/**
+			 * (Optional) List of custom actions to include in modal dialog alongwith a 'Cancel' button.
+			 * If custom 'actions' are not provided, 'OK' and 'Cancel' buttons will be shown by default.
```

Comment:
for the designed scenario of this dialog, what does it mean to have a cancel button? I think a button with 'OK' is enough, just like VSCode's API, when the default button is pressed, the API would return undefined.

Reply:
I think that's my doc error - default behavior is only with OK button.